core/image/etc/etc2_compressor.cpp

```diff
@@ -0,0 +1,148 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <assert.h>
+#include <stdlib.h>
+#include <string.h>
+#include "etc2.h"
+#include "third_party/etc2comp/EtcLib/Etc/EtcImage.h"
+
+static_assert(sizeof(etc_error) >= sizeof(Etc::Image::EncodingStatus),
+              "astc_error should superset of astcenc_error");
+
+namespace {
+const int64_t ETC_ALLOCATION_FAILED = -1;
+const uint32_t MIN_JOBS = 8;
+const uint32_t MAX_JOBS = 1024;
+const float EFFORT = 10.0f;
+const Etc::ErrorMetric ERROR_METRIC = Etc::ErrorMetric::NUMERIC;
+}  // namespace
+
+Etc::Image::Format convert_etc_format(etc_format format) {
+  switch (format) {
+    case ETC2_RGB_U8_NORM:
+      return Etc::Image::Format::RGB8;
+    case ETC2_RGBA_U8_NORM:
+      return Etc::Image::Format::RGBA8;
+    case ETC2_RGBA_U8U8U8U1_NORM:
+      return Etc::Image::Format::RGB8A1;
+    case ETC2_SRGB_U8_NORM:
+      return Etc::Image::Format::SRGB8;
+    case ETC2_SRGBA_U8_NORM:
+      return Etc::Image::Format::SRGBA8;
+    case ETC2_SRGBA_U8U8U8U1_NORM:
+      return Etc::Image::Format::SRGB8A1;
+    case ETC2_R_U11_NORM:
+      return Etc::Image::Format::R11;
+    case ETC2_RG_U11_NORM:
+      return Etc::Image::Format::RG11;
+    case ETC2_R_S11_NORM:
+      return Etc::Image::Format::SIGNED_R11;
+    case ETC2_RG_S11_NORM:
+      return Etc::Image::Format::SIGNED_RG11;
+    case ETC1_RGB_U8_NORM:
+      return Etc::Image::Format::ETC1;
+    default:
+      return Etc::Image::Format::UNKNOWN;
+  }
+}
+
+void read_image(const uint8_t* input_image, uint32_t width, uint32_t height,
+                Etc::ColorFloatRGBA* output) {
+  const uint8_t BYTE_PER_PIXEL = 4;
+  for (uint32_t h = 0; h < height; ++h) {
+    const uint8_t* src = &input_image[(h * width) * BYTE_PER_PIXEL];
+    Etc::ColorFloatRGBA* dst = &output[h * width];
+    for (uint32_t w = 0; w < width; ++w) {
+      *dst =
+          Etc::ColorFloatRGBA::ConvertFromRGBA8(src[0], src[1], src[2], src[3]);
+      dst++;
+      src += BYTE_PER_PIXEL;
+    }
+  }
+}
+
+extern "C" etc_error compress_etc(const uint8_t* input_image,
+                                  uint8_t* output_image, uint32_t width,
+                                  uint32_t height, etc_format format) {
+  auto* source_image = new Etc::ColorFloatRGBA[width * height];
+  if (!source_image) {
+    return ETC_ALLOCATION_FAILED;
+  }
+  read_image(input_image, width, height, source_image);
+  Etc::Image image(reinterpret_cast<float*>(source_image), width, height,
+                   ERROR_METRIC);
+  image.m_bVerboseOutput = false;
+
+  auto image_format = convert_etc_format(format);
+  if (image_format == Etc::Image::Format::UNKNOWN) {
+    return static_cast<etc_error>(
+        Etc::Image::EncodingStatus::ERROR_UNKNOWN_FORMAT);
+  }
+
+  auto status =
+      image.Encode(image_format, ERROR_METRIC, EFFORT, MIN_JOBS, MAX_JOBS);
+  // We don't need to care about warnings.
```

Comment:
Because we do not display or export the image in any way and it does not give any value to user to show for now.  We are trying to measure the performance difference as we can. If the user is convinced to compressing textures based on our numbers, then they can use the actual tool and depending on their compression, they may or may not get the warnings and take the proper decision for it.

Reply:
Cool, let's put this reason in the code comment! e.g.:

```
// Compression is used in replay experiments, and eventually game developers
// would compress textures outside of AGI, hence we can ignore compression
// warnings here.
```